skbio/tree/_tree.py

```diff
@@ -617,6 +617,72 @@ def lowest_common_ancestor(self, tipnames):
 
     lca = lowest_common_ancestor  # for convenience
 
+    def path(self, other, include_ends=False):
+        r"""Return the list of node names in the path from one node to another.
+
+        Parameters
+        ----------
+        other : TreeNode
+            Final node of path.
+        include_ends: bool, optional
+            Whether to include the initial and final nodes in the list.
+
+        Returns
+        -------
+        list
+            List of names of TreeNode objects.
+
+        Examples
+        --------
+        >>> from skbio import TreeNode
+        >>> tree = TreeNode.read(["((a,b)c,(d,e)f)root;"])
+        >>> node_1, node_2 = tree.find('a'), tree.find('d')
+        >>> path = node_1.path(node_2)
+        >>> print(path)
+        ['c', 'root', 'f']
+        >>> path_2 = node_1.path(node_2, include_ends=True)
+        >>> print(path_2)
+        ['a', 'c', 'root', 'f', 'd']
+
+        """
+        # find root
+        root = self.root()
+
+        # find the lowest common ancestor if not ancestors of each other
+        lca = (
```

Comment:
There are two `ancestor` and one `lca` calls, which are expensive. I tend to think that we can identify the LCA of the two nodes from their ancestors, without explicitly calling `lca`. Specifically, you can interate over the two list of ancestors from root, and get the last pair of elements that are identical. For example:

```python
anc1 = self.ancestors()
anc2 = other.ancestors()
for n1, n2 in zip(reversed(anc1), reversed(anc2)):
    if n1 is n2:
        lca = n1
    else:
        break
```